exporters/failure/collector_jira.py

```diff
@@ -41,108 +38,128 @@
+            )
 
-    def search_issues(self):
-        jira = self._connect_to_jira()
+    def _parse_issue(self, issue: Issue) -> TrackerIssue:
+        """Parse issue collected from JIRA."""
+        logging.debug(issue)
+        logging.debug(
+            "Found issue opened: %s, %s: %s",
+            issue.fields.created,
+            issue.key,
+            issue.fields.summary,
+        )
+        # Create the JiraFailureMetric
+        created_tz = parse_tz_aware(issue.fields.created, _DATETIME_FORMAT)
+        created_ts = second_precision(created_tz).timestamp()
+        resolution_ts = self._get_resolved_timestamp(issue, self.jira_resolved_statuses)
+        return TrackerIssue(
+            issue.key, created_ts, resolution_ts, self.get_app_name(issue)
+        )
 
-        critical_issues = []
+    def search_issues(self) -> List[Optional[TrackerIssue]]:
+        """
+        Search for the matching issues in JIRA.
 
+        Returns
+        -------
+        List[Optional[TrackerIssue]]
+            A list with the issues, if no error occurs; else, an empty list.
+        """
         try:
-            logging.debug("JIRA JQL query: %s" % self.jql_query_string)
-            jira_issues = []
-            start_at = 0
-            while True:
-                jira_search_results = jira.search_issues(
-                    self.jql_query_string,
-                    startAt=start_at,
-                    maxResults=JIRA_SEARCH_RESULTS,
-                    fields=self.query_result_fields_string,
-                )
-                jira_issues += jira_search_results.iterable
-                start_at += JIRA_SEARCH_RESULTS
-                logging.info("Getting jira results: %s" % start_at)
-                if start_at >= jira_search_results.total:
-                    break
-
-            for issue in jira_issues:
-                logging.debug(issue)
-                logging.debug(
-                    "Found issue opened: {}, {}: {}".format(
-                        str(issue.fields.created), issue.key, issue.fields.summary
-                    )
-                )
-                # Create the JiraFailureMetric
-                created_tz = parse_tz_aware(issue.fields.created, _DATETIME_FORMAT)
-                created_ts = second_precision(created_tz).timestamp()
-                resolution_ts = self._get_resolved_timestamp(
-                    issue, self.jira_resolved_statuses
-                )
-                tracker_issue = TrackerIssue(
-                    issue.key, created_ts, resolution_ts, self.get_app_name(issue)
-                )
-                critical_issues.append(tracker_issue)
+            jira_client = self._connect_to_jira()
+            self._set_jql_queries_strings(jira_client)
+            logging.debug("JIRA JQL query: %s", self.jql_query_string)
+            jira_issues = jira_client.search_issues(
+                self.jql_query_string,
+                startAt=0,
+                maxResults=False,
```

Comment:
I looked through the JIRA python library documentation and code, and I believe this way is better. I compared the time execution this way and the old way and it was about 0.5 seconds faster (2.9 against 3.4, approximately)

Reply:
This whole section is much cleaner, thank you.

I just want to note for later (not asking you to change this): if someone has a _ton_ of issues (relatable), then the code will look like it's hanging. The old way had some log messages interspersed. If this becomes an issue later, we could make a small abstraction over the pagination ourselves. No need to complicate things now, though!